cache.py

```diff
@@ -53,12 +51,12 @@ def datetime_encode_set(
     path = self._getvaluepath(key)
     try:
         path.parent.mkdir(exist_ok=True, parents=True)
-    except (IOError, OSError):
```

Comment:
Why was `IOError` removed from these except's?

Reply:
Because `IOError` is now a subclass of `OSError`, thus is redundant (this was a [change to the stdlib](https://docs.python.org/3/library/exceptions.html#OSError) with python 3.3 -- see the note at the end of the linked section).  This is from the pyupgrade tool, and is a great example of one of it's trade offs.  It's nice it converts to f-strings, but I agree this one is arguable (until they deprecate `IOError`.  Certainly it can be backed out, and maybe even disabled in the tool (I haven't looked that deep yet).

An excellent discussion point. :)